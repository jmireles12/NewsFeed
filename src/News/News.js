import React, { Component } from 'react'
import ApiContext from '../ApiContext'
import Articles from '../Articles/Articles'
import './News.css'

export default class News extends Component {

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;

    render() {
        const { news=[] } = this.context
        return (
            <section className='headlines'>
                <ul className='headline-list'>
                    {news.map(article =>
                        <li className='headline__li' key={article.id}>
                            <Articles
                                id={article.id}
                                title={article.fields.headline}
                                body={article.fields.bodyText}
                                publication={article.webPublicationDate}
                                url={article.webUrl}
                                thumbnail={article.fields.thumbnail}
                            />
                        </li>)}
                </ul>
            </section>
        )
    }
}