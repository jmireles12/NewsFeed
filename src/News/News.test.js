import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import News from './News'

describe(`News component`, () => {
    it('renders the complete News', () => {
        const wrapper = shallow(<News />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})