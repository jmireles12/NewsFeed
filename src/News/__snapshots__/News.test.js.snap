// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News component renders the complete News 1`] = `
<section
  className="headlines"
>
  <ul
    className="headline-list"
  />
</section>
`;
